tests/functional/test_cluster_functions.py

```diff
@@ -9,443 +9,222 @@
-                    proto_id = cluster.prototype_id
-                actual.reread()
-                assert actual.id == cluster_id
-                assert actual.prototype_id == proto_id
-
-    def test_create_cluster_with_description(self, sdk_client_fs: ADCMClient):
-        bundle = sdk_client_fs.upload_from_fs(get_data_dir(__file__, 'cluster_simple'))
-        actual = bundle.cluster_create(utils.random_string(), "description")
-        with allure.step('Check cluster list with description'):
-            cluster_list = bundle.cluster_list()
-            cluster = None
-            cluster_id = None
-            proto_id = None
-            if cluster_list:
-                for cluster in cluster_list:
-                    cluster_id = cluster.id
-                    proto_id = cluster.prototype_id
-                actual.reread()
-                assert actual.id == cluster_id
-                assert actual.prototype_id == proto_id
-                assert cluster.description == 'description'
-
-    def test_shouldnt_create_duplicate_cluster(self, sdk_client_fs: ADCMClient):
-        bundle = sdk_client_fs.upload_from_fs(get_data_dir(__file__, 'cluster_simple'))
-        bundle.cluster_create("duplicate")
-        with pytest.raises(coreapi.exceptions.ErrorMessage) as e:
-            bundle.cluster_create("duplicate")
-        with allure.step('Check error: duplicate cluster'):
-            err.CLUSTER_CONFLICT.equal(e, 'duplicate cluster')
-
-    def test_shouldnt_create_cluster_wo_proto(self, client):
-        with pytest.raises(coreapi.exceptions.ParameterError) as e:
-            client.cluster.create(name=utils.random_string())
-        with allure.step('Check error about required parameter prototype_id'):
-            assert str(e.value) == "{'prototype_id': 'This parameter is required.'}"
-        steps.delete_all_data(client)
-
-    def test_shouldnt_create_cluster_w_blank_prototype(self, client):
-        with pytest.raises(coreapi.exceptions.ErrorMessage) as e:
-            client.cluster.create(prototype_id='', name=utils.random_string())
-        with allure.step('Check error about valid integer'):
-            assert e.value.error.title == '400 Bad Request'
-            assert e.value.error['prototype_id'][0] == 'A valid integer is required.'
-        steps.delete_all_data(client)
 
-    def test_shouldnt_create_cluster_when_proto_is_string(self, client):
-        with pytest.raises(coreapi.exceptions.ErrorMessage) as e:
-            client.cluster.create(prototype_id=utils.random_string(), name=utils.random_string())
-        with allure.step('Check error about valid integer'):
-            assert e.value.error.title == '400 Bad Request'
-            assert e.value.error['prototype_id'][0] == 'A valid integer is required.'
-        steps.delete_all_data(client)
+def _create_service(cluster_: Cluster, prototype_id):
+    """Workaround to create service without retrieving prototype"""
+    cluster_._subcall(
```

Comment:
Why do you use private methods, the cluster object has an service_add() method. Isn't it enough for you?

Reply:
Some test cases require to set config directly because `config_set` at least expect iterable.